kernel/aster-nix/src/sched/priority_scheduler.rs

```diff
@@ -17,40 +16,118 @@ pub fn init() {
 /// are always prioritized during scheduling.
 /// Normal tasks are placed in the `normal_tasks` queue and are only
 /// scheduled for execution when there are no real-time tasks.
-struct PreemptScheduler {
+struct PreemptScheduler<T: PreemptSchedEntity> {
+    run_queue: SpinLock<PreemptRunQueue<T>>,
+}
+
+trait PreemptSchedEntity {
+    const REAL_TIME_TASK_PRIORITY: u16 = 100;
+
+    fn is_real_time(&self) -> bool;
+}
```

Comment:
Actually I think we can remove all priority-related things at this point since the current scheduling policy only needs a boolean variable to indicate if a task is a real-time task or not. I decided to keep the priority things to save some efforts for possible PR of Linux-like scheduling policies in the near future and I did implement a Linux-like priority design in 1e3ea625398d383948502646273019ed72289e09. I wonder how you think of the decision. If we are to keep the priority things in codebase, I'd like to apply the changes as you advised.

Reply:
> I decided to keep the priority things to save some efforts for possible PR of Linux-like scheduling policies in the near future and I did implement a Linux-like priority design in https://github.com/asterinas/asterinas/commit/1e3ea625398d383948502646273019ed72289e09. I wonder how you think of the decision.

I roughly scan the change in [1e3ea62](https://github.com/asterinas/asterinas/commit/1e3ea625398d383948502646273019ed72289e09https://github.com/asterinas/asterinas/commit/1e3ea625398d383948502646273019ed72289e09https://github.com/asterinas/asterinas/commit/1e3ea625398d383948502646273019ed72289e09https://github.com/asterinas/asterinas/commit/1e3ea625398d383948502646273019ed72289e09), do you want to move the priority to `Thread`? If so, I think it's acceptable to remove all priority-related things in OSTD.